### PR TITLE
tracing/trace_collector/library_config: updated propagation in go tracer

### PR DESCRIPTION
**Pull request by dianashevchenko**
### What does this PR do?

Updates trace context propagation docs in Go.
-  Includes **new** W3c and B3 single header styles. 
- Includes an option to disable propagation. 

### Motivation

Relates to changes in [dd-trace-go PR]( https://github.com/DataDog/dd-trace-go/pull/1630)

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/library_config/go.md [hunk 1]
```diff
@@ -118,20 +118,26 @@ Datadog may collect [environmental and diagnostic information about your system]
 
 The [APM environment name][7] may be configured [in the Agent][8] or using the [WithEnv][3] start option of the tracer.
 
-## B3 headers extraction and injection
+## Trace context propagation for distributed tracing
 
-The Datadog APM tracer supports [B3 headers extraction][9] and injection for distributed tracing.
+The Datadog APM tracer supports [B3 headers][9], [W3C][14] extraction and injection for distributed tracing.
```
Comment (urseberry) [inline content/en/tracing/trace_collection/library_config/go.md:123]:
```suggestion
The Datadog APM tracer supports [B3 headers][9], [W3C][14] extraction, and injection for distributed tracing.
```
The correct grammar here depends on what the items are. My edit assumes that the tracer supports:
- B3 headers
- W3C extraction
- injection for distributed tracing

If the words are arranged or grouped differently, let me know and can give you the appropriate punctuation. Or to make it perfectly clear, you can change this sentence to use a bulleted list.

Comment (urseberry) [reply content/en/tracing/trace_collection/library_config/go.md:123]: Maybe it's two items? In that case, it should be:

The Datadog APM tracer supports [B3 headers][9] and [W3C][14] extraction and injection for distributed tracing.

In that case, the bulleted list would be clearer.

The Datadog APM tracer supports the following for distributed tracing:
- [B3 headers][9]
- [W3C][14] extraction and injection

File: content/en/tracing/trace_collection/library_config/go.md [hunk 2]
```diff
@@ -118,20 +118,26 @@ Datadog may collect [environmental and diagnostic information about your system]
...
-configuring injection/extraction styles. Two styles are
-supported: `Datadog` and `B3`.
+configuring injection/extraction styles. Supported styles are:
+`tracecontext`, `Datadog`, [`B3`][9] and `B3 single header`.
 
-- Configure injection styles using the `DD_PROPAGATION_STYLE_INJECT=Datadog,B3` environment variable
-- Configure extraction styles using the `DD_PROPAGATION_STYLE_EXTRACT=Datadog,B3` environment variable
+- Configure injection styles using the `DD_PROPAGATION_STYLE_INJECT=tracecontext,B3` environment variable
+- Configure extraction styles using the `DD_PROPAGATION_STYLE_EXTRACT=tracecontext,B3` environment variable
+- Configure both injection and extraction styles using the `DD_TRACE_PROPAGATION_STYLE=tracecontext,B3` environment variable
```
Comment (urseberry) [inline content/en/tracing/trace_collection/library_config/go.md:131]:
```suggestion
- Configure injection styles using the `DD_PROPAGATION_STYLE_INJECT=tracecontext,B3` environment variable.
- Configure extraction styles using the `DD_PROPAGATION_STYLE_EXTRACT=tracecontext,B3` environment variable.
- Configure both injection and extraction styles using the `DD_TRACE_PROPAGATION_STYLE=tracecontext,B3` environment variable.
```

File: content/en/tracing/trace_collection/library_config/go.md [hunk 3]
```diff
@@ -118,20 +118,26 @@ Datadog may collect [environmental and diagnostic information about your system]
...
 The values of these environment variables are comma-separated lists of
-header styles enabled for injection or extraction. By default, only
-the `Datadog` extraction style is enabled.
+header styles enabled for injection or extraction. By default,
+the `tracecontext,Datadog` styles are enabled.
+
+To disable trace context propagation, set the value of the environment variables to `none`.
+- Disable injection styles using the `DD_PROPAGATION_STYLE_INJECT=none` environment variable
+- Disable extraction styles using the `DD_PROPAGATION_STYLE_EXTRACT=none` environment variable
+- Disable trace context propagation using the `DD_PROPAGATION_STYLE=none` environment variable
```
Comment (urseberry) [inline content/en/tracing/trace_collection/library_config/go.md:140]:
```suggestion
- Disable injection styles using the `DD_PROPAGATION_STYLE_INJECT=none` environment variable.
- Disable extraction styles using the `DD_PROPAGATION_STYLE_EXTRACT=none` environment variable.
- Disable trace context propagation using the `DD_PROPAGATION_STYLE=none` environment variable.
```

File: content/en/tracing/trace_collection/library_config/go.md (hunk 1 above)
Comment (katiehockman) [reply content/en/tracing/trace_collection/library_config/go.md:123]: We support both injection and extraction of both B3 headers and W3C headers. So probably can just be:

"The Datadog APM tracer supports extraction and injection of [B3][9] and [W3C][14] headers for distributed tracing."

Comment (urseberry) [reply content/en/tracing/trace_collection/library_config/go.md]: Thank you, Katie. That's perfect.

File: content/en/tracing/trace_collection/library_config/go.md [hunk 4]
```diff
@@ -118,20 +118,29 @@ Datadog may collect [environmental and diagnostic information about your system]
...
+## Trace context propagation for distributed tracing
 
-The Datadog APM tracer supports [B3 headers extraction][9] and injection for distributed tracing.
+The Datadog APM tracer supports extraction and injection of [B3][9] and [W3C][14] headers for distributed tracing.
 
 Distributed headers injection and extraction is controlled by
-configuring injection/extraction styles. Two styles are
-supported: `Datadog` and `B3`.
+configuring injection/extraction styles. Supported styles are:
+`tracecontext`, `Datadog`, [`B3`][9] and `B3 single header`.
```
Comment (urseberry) [inline content/en/tracing/trace_collection/library_config/go.md:127]:
```suggestion
`tracecontext`, `Datadog`, [`B3`][9], and `B3 single header`.
```

**Review by urseberry — APPROVED** [review]

**Pull request merged**